jack-and-the-beans/buddies-ios
# BUD-146 block users

This PR introduces client-side changes to support blocking and banning users.

There are a lot of cases you can test. For example, you can test reporting an activity that you've joined, and one that you haven't joined. Note that it may take a few seconds after reporting for changes to happen.

Also, I tried to cover the User and Activity models with a lot of tests, so that should maintain a bit of coverage.

### To Test:
1. Try reporting an activity. After you do so, you should no longer be able to see the activity anywhere. (And, from the POV of the people in the activity, you should no longer be a member)
2. Try reporting a user from an activity you are not in. After you do so, you should no longer be able to see any activities in which that user appears.
3. Try reporting a user from an activity you've joined. You should no longer be able to see any activities in which that user appears _and_ you should be removed from all of the activities you shared with that person.
4. With two different accounts, test banning a user. One account should own the activity, and the other account should join it. The owner should remove the user, and the removed user should _immediately_ see the "you can't see this activity" message while they are looking at the activity.
